ORM/DataObject.php

```diff
@@ -677,7 +706,8 @@ public function i18n_pluralise($count, $prependNumber = true) {
 	 */
 	public function singular_name() {
 		if(!$name = $this->stat('singular_name')) {
-			$name = ucwords(trim(strtolower(preg_replace('/_?([A-Z])/', ' $1', $this->class))));
+			$reflection = new \ReflectionClass($this);
```

Comment:
It's a FQN so it's not missing. We don't strictly require use statements; Using the FQN is fine.